### PR TITLE
Adds `go-sdk` to sdk test github workflow with corresponding commands

### PR DESCRIPTION
**Pull request by JoshVanL**
Branched from #6304, part of https://github.com/dapr/dapr/issues/5797 

Job can be manually triggered on a PR with the commands `/test-sdk-all` or `test-sdk-go`.

### CODE REVIEW

File: .github/workflows/dapr-test-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+          ./dist/linux_amd64/release/placement --healthz-port 9091 &
+      - name: Install dependencies
+        run: |
+          cd python-sdk
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine tox
+      - name: Check Python Examples
+        run: |
+          cd python-sdk || true
+          tox -e examples
```
Comment (artursouza) [inline .github/workflows/dapr-test-sdk.yml:88]: Can you add reporting on test progress as PR comment? This was done in E2E tests and it looks super nice and easy to follow progress.

Comment (JoshVanL) [reply .github/workflows/dapr-test-sdk.yml:88]: Hi @artursouza, I have updated the non-draft Java and Python PR https://github.com/dapr/dapr/pull/6304/commits/b1f496f9eaaaceb9b59690e247067452501131ea

**Review by mukundansundar — APPROVED** [review]

**Comment by mukundansundar** [comment]
Not for this PR, but I believe we need to enhance running tests against SDKs. 
As it exists right now, this workflow tests the PR code against master branch of the SDK. But in certain scenarios that is not the right approach, for example running against a PR against a release branch, ideally we would need to run against the latest stable version of the SDK. 
I propose we add another parameter for the individual commands `/test-sdk-java release-1.9` so that it tests the PR against the release branch specified. For the `/test-sdk-all` the current behavior is good. 

cc @artursouza @yaron2 WDYT?

**Comment by daixiang0** [comment]
> Not for this PR, but I believe we need to enhance running tests against SDKs. As it exists right now, this workflow tests the PR code against master branch of the SDK. But in certain scenarios that is not the right approach, for example running against a PR against a release branch, ideally we would need to run against the latest stable version of the SDK. I propose we add another parameter for the individual commands `/test-sdk-java release-1.9` so that it tests the PR against the release branch specified. For the `/test-sdk-all` the current behavior is good.
> 
> cc @artursouza @yaron2 WDYT?

Agree.

**Comment by ItalyPaleAle** [comment]
This doesn't seem to be working: https://github.com/dapr/dapr/actions/runs/5293791337/jobs/9583394466

File: .github/workflows/dapr-test-sdk.yml [hunk 2]
```diff
@@ -464,3 +466,124 @@ jobs:
...
+              );
+            }
+      - name: Create PR comment
+        if: env.PR_NUMBER != ''
+        uses: artursouza/sticky-pull-request-comment@v2.2.0
+        with:
+          header: ${{ github.run_id }}-go
+          number: ${{ env.PR_NUMBER }}
+          hide: true
+          hide_classify: OUTDATED
```
Comment (ItalyPaleAle) [inline .github/workflows/dapr-test-sdk.yml:511]: Please delete these or re-running the test will create a new comment without links.

**Pull request merged**